Source/SwiftLintFramework/Rules/MultilineParametersBracketsRule.swift

```diff
@@ -0,0 +1,181 @@
+                )
+            }
+            """,
+            """
+            func foo<T>(param1: T, param2: String, param3: String) -> T { /* some code */ }
+            """
+        ],
+        triggeringExamples: [
+            """
+            func foo(↓param1: String, param2: String,
+                     param3: String
+            )
+            """,
+            """
+            func foo(
+                param1: String,
+                param2: String,
+                param3: String↓)
+            """,
+            """
+            class SomeType {
+                func foo(↓param1: String, param2: String,
+                         param3: String
+                )
+            }
+            """,
+            """
+            class SomeType {
+                func foo(
+                    param1: String,
+                    param2: String,
+                    param3: String↓)
+            }
+            """,
+            """
+            func foo<T>(↓param1: T, param2: String,
+                     param3: String
+            ) -> T
+            """
+        ]
+    )
+
+    public func validate(file: File) -> [StyleViolation] {
+        return violations(in: file.structure.dictionary, file: file)
+    }
+
+    private func violations(in substructure: [String: SourceKitRepresentable], file: File) -> [StyleViolation] {
+        var violations = [StyleViolation]()
+
+        // find violations at current level
+        if let kindString = substructure.kind, let kind = SwiftDeclarationKind(rawValue: kindString),
+            SwiftDeclarationKind.functionKinds.contains(kind) {
+
+            guard
+                let nameOffset = substructure.nameOffset,
+                let nameLength = substructure.nameLength,
+                let functionName = file.contents.bridge().substringWithByteRange(start: nameOffset, length: nameLength)
+            else {
+                return []
+            }
+
+            let isMultiline = functionName.contains("\n")
+
+            let parameters = substructure.substructure.filter { $0.kind == SwiftDeclarationKind.varParameter.rawValue }
+            if isMultiline && !parameters.isEmpty {
+                if let openingBracketViolation = openingBracketViolation(parameters: parameters, file: file) {
+                    violations.append(openingBracketViolation)
+                }
+
+                if let closingBracketViolation = closingBracketViolation(parameters: parameters, file: file) {
+                    violations.append(closingBracketViolation)
+                }
+            }
+        }
+
+        // find violations at deeper levels
+        for substructure in substructure.substructure {
+            violations += self.violations(in: substructure, file: file)
+        }
+
+        return violations
+    }
+
+    private func openingBracketViolation(parameters: [[String: SourceKitRepresentable]],
+                                         file: File) -> StyleViolation? {
+        if
```

Comment:
Here's another example where `guard` could be used for early return (of `nil`, in this case). Pretty much everything can be guarded in this method, leaving the return on the happy path.

```swift
guard
    let firstParamByteOffset = ....,
    let firstParamByteLength = ....,
    ...,
    case let prefix = ...,
    case let invalidPrefixRegex = ...,
    let invalidMatch = invalidPrefixRegex.firstMatch(in: prefix, options: [], range: prefix.fullNSRange) else { return nil }

return StyleViolation()
````

Reply:
Fixed.